h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/exec/XGBoostExecutor.java

```diff
@@ -0,0 +1,18 @@
+package hex.tree.xgboost.exec;
+
+import hex.tree.xgboost.util.FeatureScore;
+
+import java.util.Map;
+
+public interface XGBoostExecutor {
+    
+    byte[] setup();
+
+    void update(int treeId);
+    
+    byte[] updateBooster();
+
+    Map<String, FeatureScore> getFeatureScores();
+
+    void cleanup();
```

Comment:
I'm thinking about whether making it AutoCloseable would be beneficial. Instead of cleanup, there would be `close` method with the same purpose and the option to use `try-with-resources` on these executors will be there.

Reply:
good idea, fixed